packages/teams-js/src/public/marketplace.ts

```diff
@@ -0,0 +1,361 @@
+import { runtime } from './runtime';
+
+/**
+ * @hidden
+ * Namespace for an app to support a checkout flow by interacting with the marketplace cart in the host.
+ * @beta
+ */
+export namespace marketplace {
+  /**
+   * @hidden
+   * Represents the cart object for the app checkout flow.
+   * @beta
+   */
+  export interface Cart {
+    /**
+     * @hidden
+     * Version of the cart.
+     */
+    readonly version: CartVersion;
+    /**
+     * @hidden
+     * The uuid of the cart.
+     */
+    readonly id: string;
+    /**
+     * @hidden
+     * The cart info.
+     */
+    readonly cartInfo: CartInfo;
+    /**
+     * @hidden
+     * The cart items.
+     */
+    readonly cartItems: CartItem[];
+  }
+
+  /**
+   * @hidden
+   * Version of the cart.
+   * @beta
+   */
+  interface CartVersion {
+    /**
+     * @hidden
+     * Represents the major version number.
+     */
+    readonly majorVersion: number;
+    /**
+     * @hidden
+     * Represents the minor version number.
+     */
+    readonly minorVersion: number;
+  }
+
+  /**
+   * @hidden
+   * Represents the cart information
+   * @beta
+   */
+  interface CartInfo {
+    /**
+     * @hidden
+     * The country market where the products are selling.
+     * Should be country code in ISO 3166-1 alpha-2 format, e.g. CA for Canada.
+     * https://en.wikipedia.org/wiki/ISO_3166-1_alpha-2
+     */
+    readonly market: string;
+    /**
+     * @hidden
+     * The identifier to tell the cart is checked out by admin or end user.
+     */
+    readonly intent: Intent;
+    /**
+     * @hidden
+     * Locale the app should render for the user
+     * Should be a BCP 47 language tag, e.g. en-US ([primary tag]-[ISO 3166-1 alpha-2 code]).
+     * https://en.wikipedia.org/wiki/IETF_language_tag
+     */
+    readonly locale: string;
+    /**
+     * @hidden
+     * The status of the cart.
+     */
+    readonly status: CartStatus;
+    /**
+     * @hidden
+     * ISO 4217 currency code for the cart item price, e.g. USD for US Dollar.
+     * https://en.wikipedia.org/wiki/ISO_4217
+     */
+    readonly currency: string;
+    /**
+     * @hidden
+     * ISO 8601 timestamp string in UTC, indicates when the cart is created.
+     * e.g. 2023-06-19T22:06:59Z
+     * https://en.wikipedia.org/wiki/ISO_8601
+     */
+    readonly createdAt: string;
+    /**
+     * @hidden
+     * ISO 8601 timestamp string in UTC, indicates when the cart is updated.
+     * e.g. 2023-06-19T22:06:59Z
+     * https://en.wikipedia.org/wiki/ISO_8601
+     */
+    readonly updatedAt: string;
+  }
+
+  /**
+   * @hidden
+   * Represents the basic cart item information.
+   * @beta
+   */
+  export interface Item {
+    /**
+     * @hidden
+     * The id of the cart item.
+     */
+    readonly id: string;
+    /**
+     * @hidden
+     * The display name of the cart item.
+     */
+    readonly name: string;
+    /**
+     * @hidden
+     * The quantity of the cart item.
+     */
+    readonly quantity: number;
```

Comment:
>number [](http://example.com/codeflow?start=23&length=6)

Not sure how deep you want to go to ensuring misuse is avoided but you could consider a custom type to prevent this value from ever being negative (assuming it's invalid to have a negative quantity of items)